### ISSUE TITLE
[MA0066]: Should not trigger for `ImmutableHashSet<T>.Empty.WithComparer(...)`

### ISSUE THREAD

**Issue by Pretasoc** [meziantou/Meziantou.Analyzer#822]
### Version of the Meziantou.Analyzer NuGet package

2.0.210

### Rule Identifier

MA0066

### Target Framework

net9.0.302

### C# Language version

preview

### Description

When using an `IEqualityComparer<T>` on the empty instance of the immutable set like types (`ImmutableHashSet<T>`, `ImmutableDictionary<TKey, TValue>`) MA0066 should not trigger.

### Reproduction Steps

Minimal code:

```c#
#r "nuget: Meziantou.Analyzer, 2.0.210"

using System.Collections.Immutable;

ImmutableHashSet<KeyValuePair<string, string>> x = ImmutableHashSet<KeyValuePair<string, string>>.Empty.WithComparer(new KvComparer());
ImmutableDictionary<KeyValuePair<string, string>, string> z = ImmutableDictionary<KeyValuePair<string, string>, string>.Empty.WithComparers(new KvComparer());



public class KvComparer : IEqualityComparer<KeyValuePair<string, string>>
{
    public bool Equals(KeyValuePair<string, string> x, KeyValuePair<string, string> y)
    {
        throw new NotImplementedException();
    }
    
    public int GetHashCode(KeyValuePair<string, string> x)
    {
        throw new NotImplementedException();
    }
}

```

**Comment by meziantou** [comment]
I understand the issue and why it would be nice to fix it. However, I'm not sure this is worth the effort. I think you can use the `ImmutableXyz.Create` methods with a comparer to get a compliant code:

````c#
_ = ImmutableHashSet.Create<KeyValuePair<string, string>>(new KvComparer());
_ = ImmutableDictionary.Create<KeyValuePair<string, string>, string>(new KvComparer());
````

This seems shorter and easier to read

**Comment by Pretasoc** [comment]
I see your point. I personally prefer the `Empty.With(...)` style. but thats simply taste.

**Comment by Frulfump** [comment]
Should have been closed as not planned instead.
